desktop/libs/indexer/src/indexer/api3.py

```diff
@@ -208,11 +220,50 @@ def guess_format(request):
   format_['status'] = 0
   return JsonResponse(format_)
 
+def decode_utf8(input_iterator):
+  for l in input_iterator:
+    yield l.decode('utf-8')
 
 def guess_field_types(request):
   file_format = json.loads(request.POST.get('fileFormat', '{}'))
 
-  if file_format['inputFormat'] == 'file':
+  if file_format['inputFormat'] == 'localfile':
+    upload_file = request.FILES['inputfile']
+    fs = FileSystemStorage()
+    name = fs.save(upload_file.name, upload_file)
```

Comment:
We can see for making it temporary, handling name conflicts, user prefix maybe too in follow-ups